src/components/Api.js

```diff
@@ -0,0 +1,103 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) return res.json();
+    return res
+      .json()
+      .then((error) =>
+        Promise.reject(`Error: ${res.status} - ${error.message || error.error}`)
+      );
+  }
+
+  _request(url, options) {
+    return fetch(url, options).then(this._checkResponse);
```

Comment:
**Could be improved**


Please, use the `_request` method everywhere in `Api`


```js

  // Get Current User Info
  getProfile() {
    return this._request(this._baseUrl + "/users/me", {
      method: "GET",
      headers: this._headers,
    });
  }

  // Update Profile Info
  patchProfileInfo(nameVar, bioVar) {
    return this._request(this._baseUrl + "/users/me", {
      method: "PATCH",
      headers: this._headers,
      body: JSON.stringify({
        name: nameVar,
        about: bioVar,
      }),
    });
  }
```